components/project-list/project-list.jsx

```diff
@@ -1,113 +1,49 @@
-      entries: [],
-      displayBatchIndex: 1
-    };
-  },
   getDefaultProps() {
     return {
-      params: {
-        featured: null,
-        issue: null,
-        ids: null
-      }
+      entries: []
     };
   },
-  componentDidMount() {
-    this.fetchData(this.props.params);
-  },
-  componentWillReceiveProps(nextProps) {
-    // Reset state
-    this.resetState();
-
-    // <ProjectList> doesn't always get unmounted and remounted when navigating between pages.
-    // (e.g., When navigating between the /issue/issue-name pages <Issue><ProjectList></Issue>
-    //        do not get re-mounted since the same components are being rendered.)
-    // It is treated as passing new props to <ProjectList {...newProps} />
-    let newIssueEntered = (nextProps.params.issue !== this.props.params.issue);
-
-    if ( newIssueEntered ) {
-      // Fetch data based on the new params props to ensure data gets fetched and displayed accordingly.
-      this.fetchData(nextProps.params);
-    }
-  },
-  resetState() {
-    this.setState(this.getInitialState());
-  },
-  fetchData(params = {}, entries = [], apiPageIndex = 1) {
-    params.page = apiPageIndex;
-
-    Service.entries
-      .get(params)
-      .then((data) => {
-        entries = entries.concat(data.results);
-
-        if (data.next) { // there are more data in the database we need to fetch
-          apiPageIndex += 1;
-          this.fetchData(params);
-        } else {
-          this.setState({
-            dataLoaded: true,
-            entries: entries
-          });
-        }
-      })
-      .catch((reason) => {
-        console.error(reason);
-      });
-  },
-  applyFilterToList(entries,params = {}) {
-    if ( params.ids ) { // we want to show the list from the most recently bookmarked entry first
-      let bookmarkedIdArray = params.ids.split(`,`);
-
-      return entries.sort((a,b) => {
-        return bookmarkedIdArray.indexOf(a.id.toString()) > bookmarkedIdArray.indexOf(b.id.toString());
-      });
-    } else {
-      return entries;
-    }
-
+  propTypes: {
+    entries: React.PropTypes.array.isRequired,
+    loadingData: React.PropTypes.bool.isRequired,
+    moreEntriesToFetch: React.PropTypes.bool.isRequired,
+    fetchData: React.PropTypes.func.isRequired
   },
   handleViewMoreClick() {
-    this.setState({displayBatchIndex: this.state.displayBatchIndex+1});
+    this.props.fetchData();
+  },
+  renderProjectCards() {
+    return this.props.entries.map(project => {
+      return <ProjectCard key={project.id} {...Utility.processEntryData(project)} />;
+    });
+  },
+  renderLoadingNotice() {
+    // 3 empty <div></div> here are for the loading animation dots (done in CSS) to show.
+    return this.props.loadingData ?
+      (<div className="loading my-5 d-flex justify-content-center align-items-center">
+        <div></div>
+        <div></div>
+        <div></div>
+      </div>) : null;
+  },
+  renderViewMoreBtn() {
+    return this.props.moreEntriesToFetch ?
```

Comment:
I'd return early, rather than use a tertiary here:
```
if (!this.props.moreEntriesToFetch) return null;
return (
  <div>
    ...
  </div>
);
```